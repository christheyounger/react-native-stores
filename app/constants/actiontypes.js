export const GET_STORES = "GET_STORES";
export const GOT_STORES = "GOT_STORES";
export const GET_STORES_FAIL = "GET_STORES_FAIL";
export const GET_STORE = "GET_STORE";
export const GOT_STORE = "GOT_STORE";
export const GET_STORE_FAIL = "GET_STORE_FAIL";
export const GET_STORE_BRANCHES = "GET_STORE_BRANCHES";
export const GOT_STORE_BRANCHES = "GOT_STORE_BRANCHES";
export const GET_STORE_BRANCHES_FAIL = "GET_STORE_BRANCHES_FAIL";
