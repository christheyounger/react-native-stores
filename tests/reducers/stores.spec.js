import reducer from '../../app/reducers/stores';
import { GET_STORES, GOT_STORES } from '../../app/constants/actiontypes';

describe('reducer.stores', () => {
  test('should GET_STORES', () => {
    expect(reducer(null, {type: GET_STORES})).toMatchSnapshot();
  });
  test('should GOT_STORES', () => {
    const result = reducer(
        {loading: true, stores: []},
        {type: GOT_STORES, payload: {stores: ['store 1']}}
    );
    expect(result).toMatchSnapshot();
  });
  test('should have a default', () => {
    const state = {cow: 'horse'};
    expect(reducer(state, {type: 'pony'})).toEqual(state);
  });
});
