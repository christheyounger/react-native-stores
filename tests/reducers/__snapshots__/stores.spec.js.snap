// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer.stores should GET_STORES 1`] = `
Object {
  "loading": true,
  "stores": Array [],
}
`;

exports[`reducer.stores should GOT_STORES 1`] = `
Object {
  "loading": false,
  "stores": Array [
    "store 1",
  ],
}
`;
